zap/src/main/java/org/zaproxy/zap/scan/filters/FilterResult.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.scan.filters;
+
+/** @author KSASAN preetkaran20@gmail.com */
+public class FilterResult {
```

Comment:
This should be immutable (like `ParseFilter.FilterResult`), that way we can reuse/cache the results.

Reply:
hmm. Got your point. will update it.

Reply:
The instance variables could be final and the no-arg constructor private (the class should expose an instance for non-filtered result).

Reply:
sure will do the changes.